x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutJobAction.java

```diff
@@ -15,38 +18,85 @@
+import org.elasticsearch.xpack.core.security.SecurityContext;
+import org.elasticsearch.xpack.ml.datafeed.DatafeedManager;
 import org.elasticsearch.xpack.ml.job.JobManager;
 
 public class TransportPutJobAction extends TransportMasterNodeAction<PutJobAction.Request, PutJobAction.Response> {
 
+    private static final Logger logger = LogManager.getLogger(TransportPutJobAction.class);
     private final JobManager jobManager;
+    private final DatafeedManager datafeedManager;
     private final XPackLicenseState licenseState;
     private final AnalysisRegistry analysisRegistry;
+    private final SecurityContext securityContext;
+
 
     @Inject
-    public TransportPutJobAction(TransportService transportService, ClusterService clusterService,
+    public TransportPutJobAction(Settings settings, TransportService transportService, ClusterService clusterService,
                                  ThreadPool threadPool, XPackLicenseState licenseState, ActionFilters actionFilters,
                                  IndexNameExpressionResolver indexNameExpressionResolver, JobManager jobManager,
-                                 AnalysisRegistry analysisRegistry) {
+                                 DatafeedManager datafeedManager, AnalysisRegistry analysisRegistry) {
         super(PutJobAction.NAME, transportService, clusterService, threadPool, actionFilters, PutJobAction.Request::new,
             indexNameExpressionResolver, PutJobAction.Response::new, ThreadPool.Names.SAME);
         this.licenseState = licenseState;
         this.jobManager = jobManager;
         this.analysisRegistry = analysisRegistry;
+        this.datafeedManager = datafeedManager;
+        this.securityContext = XPackSettings.SECURITY_ENABLED.get(settings) ?
+            new SecurityContext(settings, threadPool.getThreadContext()) : null;
     }
 
     @Override
     protected void masterOperation(Task task, PutJobAction.Request request, ClusterState state,
                                    ActionListener<PutJobAction.Response> listener) throws Exception {
-        jobManager.putJob(request, analysisRegistry, state, listener);
+        jobManager.putJob(request, analysisRegistry, state, ActionListener.wrap(
+            jobCreated -> {
+                if (jobCreated.getResponse().getDatafeedConfig().isPresent() == false) {
+                    listener.onResponse(jobCreated);
+                    return;
+                }
+                datafeedManager.putDatafeed(
+                    new PutDatafeedAction.Request(jobCreated.getResponse().getDatafeedConfig().get()),
+                    // Use newer state from cluster service as the job creation may have created shared indexes
+                    clusterService.state(),
+                    licenseState,
+                    securityContext,
+                    threadPool,
+                    ActionListener.wrap(
+                        created -> listener.onResponse(jobCreated),
```

Comment:
```suggestion
                        datafeedCreated -> listener.onResponse(jobCreated),
```
?
This will instantly give the reader a clue what is created.